src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -124,21 +124,75 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
     }
 
     qint64 offset = 0;
+    QByteArray protrudingData;
+    bool protrudingDataLeft = false;
 
     QObject::connect(&socket, &QLocalSocket::readyRead, &loop, [&] {
-        auto data = socket.readAll();
+        const auto data = socket.readAll();
         if (data.isEmpty()) {
             qCWarning(lcCfApiWrapper) << "Unexpected empty data received" << requestId;
             sendTransferError();
             loop.quit();
             return;
         }
-        sendTransferInfo(data, offset);
-        offset += data.length();
+
+        QByteArray dataToSent;
+        // CFAPI expectes sended blocks to be of a specified size.
+        // Only the last sended block is allowed to be smaller than the specified block size
+        // and therefore marks the end of the transmission.
+        static constexpr auto CFAPI_BLOCK_SIZE = 4096;
+
+        int dataIndex = 0;
+        // Is there any data left from last run?
+        if (!protrudingData.isEmpty()) {
+            QByteArray dataToSent;
+            dataToSent.append(protrudingData);
+
+            // Fill up remaining first block if possible
+            const auto fillupSize = CFAPI_BLOCK_SIZE - dataToSent.size();
+            if (fillupSize > 0) {
+                dataToSent.append(data.left(fillupSize));
```

Comment:
@FlexW Can we safely assume that **fillupSize** will never exceed the `data.size()`? Does it make sense to change it to the following?
`const auto missingSize = qMin(fillupSize, data.size());
dataToSent.append(data.left(missingSize));
dataIndex += missingSize;`

Reply:
Yes I think this could happen. Will change it:)

Reply:
@allexzander But I think the `if` statement is still required in case `fillupSize` is negative.

Reply:
> @allexzander But I think the `if` statement is still required in case `fillupSize` is negative.

Makes sense.